File: wstikkelorum/src/ast/expression/Eq.java
package ast.expression;

import ast.Visitor;

public class Eq extends Expression {
	private Expression lhs;
	private Expression rhs;
	
	public Eq(Expression result, Expression result2) {
		this.lhs = result;
		this.rhs = result2;
	}
	
	public Expression getLhs() {
		return lhs;
	}

	public Expression getRhs() {
		return rhs;
	}
	
	@Override
	public void accept(Visitor visitor) {
		visitor.visit(this);
	}

}

File: Edward Poot/src/ql/ast/expr/binaryoperatorexpr/eq.go
package binaryoperatorexpr

import (
	"ql/ast/expr"
	"ql/ast/visit"
)

type Eq struct {
	Lhs, Rhs expr.Expr
}

func (e Eq) GetLhs() expr.Expr {
	return e.Lhs
}

func (e Eq) GetRhs() expr.Expr {
	return e.Rhs
}

func (e Eq) Eval(s interface{}) interface{} {
	switch e.Lhs.Eval(s).(type) {
	case int:
		return e.GetLhs().Eval(s).(int) == e.GetRhs().Eval(s).(int)
	case bool:
		return e.GetLhs().Eval(s).(bool) == e.GetRhs().Eval(s).(bool)
	default:
		panic("Eq error: comparing unknown types")
	}
}

func (e Eq) Accept(v visit.Visitor, s interface{}) interface{} {
	return v.Visit(e, s)
}